js/customize-migrate.js

```diff
@@ -0,0 +1,46 @@
+/* global jQuery, wp */
+(function( $ ) {
+	'use strict';
+	var component = {
+		doingAjax: false,
+		postMigrationCount: 20
+	};
+	component.init = function() {
+		$( function() {
+			component.el = $( '#customize-snapshot-migration' );
+			component.bindClick();
+			component.spinner = $( '.spinner.customize-snapshot-spinner' );
+			component.spinner.css( 'margin', '0' );
+		} );
+	};
+	component.bindClick = function() {
+		component.el.click( function() {
+			if ( component.doingAjax ) {
+				return;
+			}
+			component.spinner.css( 'visibility', 'visible' );
+			component.doingAjax = true;
+			component.migrate( component.el.data( 'nonce' ), component.postMigrationCount );
+		} );
+	};
+	component.migrate = function( nonce, limit ) {
+		var requestData, request;
+		requestData = {
+			nonce: nonce,
+			limit: limit
+		};
+		request = wp.ajax.post( 'customize_snapshot_migration', requestData );
+		request.always( function( data ) {
```

Comment:
I will change it to `request.done` and will terminate ajax loop in case of fail.